.github/workflows/lintandbuild.yml

```diff
@@ -0,0 +1,31 @@
+#
+# This source file is part of the ENGAGE-HF based on the Stanford Spezi Template Application project
+#
+# SPDX-FileCopyrightText: 2023 Stanford University
+#
+# SPDX-License-Identifier: MIT
+#
+
+name: Lint and Build
+
+on: [pull_request]
+
+jobs:
+  deploy:
+    name: Lint and Build Functions
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```

Comment:
Would be good to use v4 and generally check that we use the latest version of the actions.

Reply:
updated both to latest major version (actions/checkout and actions/setup-node)